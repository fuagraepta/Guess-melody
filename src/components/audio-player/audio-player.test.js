import React from 'react';
import renderer from 'react-test-renderer';

import AudioPlayer from './audio-player.jsx';

const mockTrackSrc = `https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg`;
const mockElement = <audio/>;

describe(`The component is rendered correctly.`, () => {
  it(`Audio player correctly render with transferred mock track.`, () => {
    const audioPlayer = renderer.
    create(
        <AudioPlayer
          isPlaying={true}
          isLoading={true}
          src = {mockTrackSrc}
          onPlayButtonClick = {jest.fn()}>
          {mockElement}
        </AudioPlayer>
    )
    .toJSON();

    expect(audioPlayer).toMatchSnapshot();
  });
});
