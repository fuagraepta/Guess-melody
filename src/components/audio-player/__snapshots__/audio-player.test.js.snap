// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly Audio player correctly render with transferred mock track 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    disabled={true}
    onClick={[MockFunction]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
